container_pipeline/workers/dockerfile_lint.py

```diff
@@ -0,0 +1,131 @@
+            logger.info("/tmp/scan directory already exists")
+        elif os.path.isfile("/tmp/scan"):
+            os.remove("/tmp/scan")
+            os.makedirs("/tmp/scan")
+        else:
+            os.makedirs("/tmp/scan")
+
+        with open("/tmp/scan/Dockerfile", "w") as f:
+            f.write(dockerfile)
+
+    def lint(self, job):
+        """
+        Lint the Dockerfile received
+        """
+        out, err = subprocess.Popen(
+            ["docker",
+             "run",
+             "--rm",
+             "-v",
+             "/tmp/scan:/root/scan:Z",
+             "registry.centos.org/pipeline-images/dockerfile-lint"],
+            stdout=subprocess.PIPE
+        ).communicate()
+
+        if err is None:
+            logger.info("Dockerfile Lint check done. Exporting logs.")
+            # logs file for linter
+            logs_file_path = os.path.join(
+                job.get("logs_dir"),
+                settings.LINTER_RESULT_FILE
+
+            )
+
+            # logs URL for linter results
+            logs_URL = logs_file_path.replace(
+                settings.LOGS_DIR_BASE,   # /srv/pipeline-logs/
+                settings.LOGS_URL_BASE   # https://registry.centos.org
+            )
+
+            out += "\nHosted linter results : %s\n" % logs_URL
+            # export linter results
+            self.export_logs(out, logs_file_path)
+
+            response = {
+                "logs": out,
+                "linter_results": True,
+                "action": "notify_user",
+                "namespace": job.get('namespace'),
+                "notify_email": job.get("notify_email"),
+                "job_name": job.get("job_name"),
+                "msg": None,
+                "linter_results_path": logs_file_path,
+                "logs_URL": logs_URL,
+            }
+
+        else:
+            logger.error(
+                "Dockerfile Lint check failed", extra={'locals': locals()})
+            response = {
+                "linter_results": False,
+                "action": "notify_user",
+                "namespace": job.get('namespace'),
+                "notify_email": job.get("notify_email"),
+                "job_name": job.get("job_name"),
+                "msg": err,
+            }
+
+        # linter execution status file path
+        status_file_path = os.path.join(
+            job.get("logs_dir"),
+            settings.LINTER_STATUS_FILE
+        )
+        # now export the status about linter execution in logs dir of the job
+        # this response will be read after job builds and sending email to user
+
+        # TODO: Write export JSON method in base worker
+        self.export_linter_status(response, status_file_path)
+
+    def export_linter_status(self, status, status_file_path):
+        """
+        Export status of linter execution for build in process
+        """
+        try:
+            fin = open(status_file_path, "w")
+            json.dump(status, fin)
```

Comment:
A file opened for any purpose should be closed as well, right? Also, how about using `with open` instead?